profiler/profiler.go

```diff
@@ -189,6 +195,30 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 	}
 }
 
+// enabledProfileTypes returns the enabled profile types in a deterministic
+// order. The CPU profile always comes first because people might spot
+// interesting events in there and then try to look for the counter-part event
+// in the mutex/heap/block profile. Deterministic ordering is also important
+// for delta profiles, otherwise they'd cover varying profiling periods.
+func (p *profiler) enabledProfileTypes() []ProfileType {
+	order := []ProfileType{
```

Comment:
Regarding my previous comment about testing to ensure that `order` contains all profiles (which may be added in the future), I think one test would be to get all keys of the `profileTypes` map, and make sure they get returned here when they are all enabled.

Reply:
Yeah, that's a test case I could add 👍

Reply:
3fcf217 should cover the soundness of the `ProfileType` enum list, `enabledProfileTypes()` as well as the `profileTypes` map.